src/effects/Reverb_libSoX.h

```diff
@@ -218,6 +268,174 @@ typedef struct {
+   //        |         Imagine someone just recorded "A B C D E", and now we are going to read "A", but a resize request comes.
+   //        v         
+   //  | C B A E D |
+   //
+   
+   // Depending on the new requested size and where ptr is, we can have these cases:
+   // 
+   //        v
+   //  | C B A 0 E D |  bigger size: right-shift what is right of ptr,
+   //                   and fill the resulting gap with zeros.
+   //
+   //                   (with variant: ptr is already at the old right edge - then shift nothing, just append zeros)
+   //        v
+   //  | C B A D |      smaller size, and ptr is within it (but not at the edge): left-shift what is to the right of ptr,
+   //                   discarding samples.
+   // 
+   //        V
+   //  | C B A |        smaller size, and ptr would be at the new right edge; do nothing.
+   // 
+   //      v
+   //  | B A |          smaller size, and ptr would be beyond the new right edge: left-shift what is ahead of ptr,
+   //                   AND move ptr to size-1
+
+   assert(newSize > 0);
+
+   const int    sizeDiff = (newSize - p->size);
+   const size_t ptrPos   = (p->ptr  - p->buffer);
+
+   const size_t numSamplesBehindPtr = (p->size - 1) - ptrPos;
+
+   if (sizeDiff > 0)
+   {
+      // case: bigger size
+
+      if (numSamplesBehindPtr > 0)
+      {
+         // right-shift what is right of ptr
+         memcpy(p->ptr + 1 + sizeDiff, p->ptr + 1, numSamplesBehindPtr * sizeof(float));
+      }      
+
+      // fill the created gap with zeros
+      memset(p->ptr + 1, 0, sizeDiff * sizeof(float));
+   }
+   else if (sizeDiff < 0)
+   {
+      // case: smaller size
+            
+      if (ptrPos < newSize-1)
+      {
+         size_t lenOfBlockToShift = newSize - 1 - ptrPos;
+         float* ptrToBlockToShift = p->buffer + p->size - lenOfBlockToShift;
+         memcpy(p->ptr + 1, ptrToBlockToShift, lenOfBlockToShift * sizeof(float));         
+      }
+      else if (ptrPos == newSize - 1)
+      {
+         // sub-case: ptr is at the new edge - no shifting to do, and ptr can stay where it is
+      }
+      else
+      {
+         // sub-case: ptr would be beyond the new edge
+         // left-shift what is ahead of ptr and make ptr point to the new edge
+         size_t samplesBeyond = ptrPos - newSize + 1;
+         memcpy(p->buffer, p->ptr - newSize + 1, newSize * sizeof(float));
+         p->ptr = p->buffer + newSize - 1;
+      }
+   }
+
+   p->size = newSize;
+}
+
+
+static void filter_t_resize(filter_t* p, size_t newSize)
+{
+   // Choose your resize method.
+   //
+   // When moving the Room Size slider:
+   //
+   // - using resize_resetting, you will not hear previous reverberations,
+   //   but at the same time you will not hear sound artifacts
+   //
+   // - using resize_preserving, you will hear previous reverberations,
+   //   but sometimes you might hear sound artifacts
```

Comment:
Basically the story is: I started doing the more complicate thing (i.e. preserving as much as possible of what is in the comb/allpass filters) and that ended up having sound artifacts, whereas the simple solution had no artifacts (but then users will hear no reverberation as long as they move the slider).

About more elaborate ideas: we could cross-fade between an instance running with the last settings before a slider was moved, and an instance with the settings currently set by the sliders. But that requires double processing and further complications, I doubt that it would be worth.

Reply:
> Is the Room Size slider perhaps the only one for which it would be very importatn?

I think so.